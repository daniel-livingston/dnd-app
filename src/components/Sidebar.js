import React from "react";
import { Link } from "react-router-dom";

const Sidebar = () => (
	<div className='sidebar column-center bg-secondary'>
		<Link className='sidebar-button' to='/'>
			<div className='sidebar__item'>
				<svg viewBox='0 0 16 16' className='icon-w' xmlns='http://www.w3.org/2000/svg'>
					<path
						fillRule='evenodd'
						d='M2 13.5V7h1v6.5a.5.5 0 0 0 .5.5h9a.5.5 0 0 0 .5-.5V7h1v6.5a1.5 1.5 0 0 1-1.5 1.5h-9A1.5 1.5 0 0 1 2 13.5zm11-11V6l-2-2V2.5a.5.5 0 0 1 .5-.5h1a.5.5 0 0 1 .5.5z'
					/>
					<path
						fillRule='evenodd'
						d='M7.293 1.5a1 1 0 0 1 1.414 0l6.647 6.646a.5.5 0 0 1-.708.708L8 2.207 1.354 8.854a.5.5 0 1 1-.708-.708L7.293 1.5z'
					/>
				</svg>
				<div className='sidebar__item__title text-primary'>Home</div>
			</div>
		</Link>
		<Link className='sidebar-button' to='/canvas'>
			<div className='sidebar__item'>
				<svg className='icon-w' viewBox='0 0 16 16' xmlns='http://www.w3.org/2000/svg'>
					<path d='M8.473.337a.5.5 0 0 0-.946 0L6.954 2h2.092L8.473.337zM12.15 11h-1.058l1.435 4.163a.5.5 0 0 0 .946-.326L12.15 11zM8.5 11h-1v2.5a.5.5 0 0 0 1 0V11zm-3.592 0H3.85l-1.323 3.837a.5.5 0 1 0 .946.326L4.908 11z' />
					<path
						fillRule='evenodd'
						d='M14 3H2v7h12V3zM2 2a1 1 0 0 0-1 1v7a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1V3a1 1 0 0 0-1-1H2z'
					/>
				</svg>
				<div className='sidebar__item__title text-primary'>Canvas</div>
			</div>
		</Link>
		<Link className='sidebar-button' to='/dice'>
			<div className='sidebar__item'>
				<svg viewBox='0 0 16 16' className='icon-w' xmlns='http://www.w3.org/2000/svg'>
					<path
						fillRule='evenodd'
						d='M13 1H3a2 2 0 0 0-2 2v10a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3a2 2 0 0 0-2-2zM3 0a3 3 0 0 0-3 3v10a3 3 0 0 0 3 3h10a3 3 0 0 0 3-3V3a3 3 0 0 0-3-3H3z'
					/>
					<circle cx='4' cy='4' r='1.5' />
					<circle cx='12' cy='12' r='1.5' />
					<circle cx='8' cy='8' r='1.5' />
				</svg>
				<div className='sidebar__item__title text-primary'>Dice</div>
			</div>
		</Link>
		<Link className='sidebar-button' to='/settings'>
			<div className='sidebar__item'>
				<svg viewBox='0 0 16 16' className='icon-w' xmlns='http://www.w3.org/2000/svg'>
					<path
						fillRule='evenodd'
						d='M8.837 1.626c-.246-.835-1.428-.835-1.674 0l-.094.319A1.873 1.873 0 0 1 4.377 3.06l-.292-.16c-.764-.415-1.6.42-1.184 1.185l.159.292a1.873 1.873 0 0 1-1.115 2.692l-.319.094c-.835.246-.835 1.428 0 1.674l.319.094a1.873 1.873 0 0 1 1.115 2.693l-.16.291c-.415.764.42 1.6 1.185 1.184l.292-.159a1.873 1.873 0 0 1 2.692 1.116l.094.318c.246.835 1.428.835 1.674 0l.094-.319a1.873 1.873 0 0 1 2.693-1.115l.291.16c.764.415 1.6-.42 1.184-1.185l-.159-.291a1.873 1.873 0 0 1 1.116-2.693l.318-.094c.835-.246.835-1.428 0-1.674l-.319-.094a1.873 1.873 0 0 1-1.115-2.692l.16-.292c.415-.764-.42-1.6-1.185-1.184l-.291.159A1.873 1.873 0 0 1 8.93 1.945l-.094-.319zm-2.633-.283c.527-1.79 3.065-1.79 3.592 0l.094.319a.873.873 0 0 0 1.255.52l.292-.16c1.64-.892 3.434.901 2.54 2.541l-.159.292a.873.873 0 0 0 .52 1.255l.319.094c1.79.527 1.79 3.065 0 3.592l-.319.094a.873.873 0 0 0-.52 1.255l.16.292c.893 1.64-.902 3.434-2.541 2.54l-.292-.159a.873.873 0 0 0-1.255.52l-.094.319c-.527 1.79-3.065 1.79-3.592 0l-.094-.319a.873.873 0 0 0-1.255-.52l-.292.16c-1.64.893-3.433-.902-2.54-2.541l.159-.292a.873.873 0 0 0-.52-1.255l-.319-.094c-1.79-.527-1.79-3.065 0-3.592l.319-.094a.873.873 0 0 0 .52-1.255l-.16-.292c-.892-1.64.902-3.433 2.541-2.54l.292.159a.873.873 0 0 0 1.255-.52l.094-.319z'
					/>
					<path
						fillRule='evenodd'
						d='M8 5.754a2.246 2.246 0 1 0 0 4.492 2.246 2.246 0 0 0 0-4.492zM4.754 8a3.246 3.246 0 1 1 6.492 0 3.246 3.246 0 0 1-6.492 0z'
					/>
				</svg>
				<div className='sidebar__item__title text-primary'>Settings</div>
			</div>
		</Link>
	</div>
);

export default Sidebar;
