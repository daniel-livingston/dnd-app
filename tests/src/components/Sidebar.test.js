import React from "react";
import { mount, shallow } from "enzyme";
import { MemoryRouter, Link } from "react-router-dom";
import Sidebar from "../../../src/components/Sidebar";

const wrappedSidebar = (
	<MemoryRouter>
		<Sidebar />
	</MemoryRouter>
);

describe("<Sidebar>", () => {
	test("renders correctly", () => {
		const wrapper = shallow(<Sidebar />);
		expect(wrapper).toMatchSnapshot();
	});

	test("renders correct links", () => {
		const wrapper = mount(wrappedSidebar);
		expect(wrapper.find(Link).at(0).prop("to")).toBe("/");
		expect(wrapper.find(Link).at(1).prop("to")).toBe("/canvas");
		expect(wrapper.find(Link).at(2).prop("to")).toBe("/dice");
		expect(wrapper.find(Link).at(3).prop("to")).toBe("/settings");
	});
});
