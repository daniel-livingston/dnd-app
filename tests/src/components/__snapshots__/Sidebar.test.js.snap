// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar> renders correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "exr4ne",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "exr4ne",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Sidebar />
</Router>
`;
